### PR TITLE
Feature 190

### LINKED ISSUE #190: Once user settings are implemented add Delete confirmation Setting
Once we have user based settings add in an override for Deletion confirmation.  It is defaulted to off, but the user can enable it.

### PR DESCRIPTION
**Pull request by BryonLewis**
Fixes #190 

- Adds into the user Settings a prompt setting for prompting the user when deleting objects.
- Restructured `NewTrackSettings` into having `TrackSettings` with a new `DeletionSettings`.  It is one value for now but I don't know if more will be added in the future.
- CreationMode has been renamed to TrackSettings to more accurately reflect being tracksettings
- The localStorage uses the same format just with an additional field.
- Updated the `useModeManager` `handleRemoveTrack` to prompt if the user settings indicate so.  There is an option override which is used for  track visibility deletion tool (The delete button next to track settings).

Selection_265

- Not a huge fan of having the multiple top-level items in the storage but I didn't want to interrupt anyone's current track creation settings.

### CODE REVIEW

**Review by subdavis — CHANGES_REQUESTED** [review]
We should be vigilant about naming and avoid proliferation of symbols with names that 1) don't really describe the entity and 2) are very close to other things.  There's a similar problem in `common.ts` in desktop.

Stuff like this really impacts readability because rather than relying on intuition, you have to rely on bulk memorization to know what things do.

Please hold me accountable to this as well.

Rather than introduce 2 more symbols to the `useSettings` namespace, could we just rename `NewTrackSettings` to `TrackSettings` add `promptUserOnDelete` to `TrackSettings`.

Putting `DeletionSettings` into its own namespace doesn't make sense, because if we end up with settings that affect other types of deletions (users, types, etc) then we can't put them in DeletionSettings because it's a child of `TrackSettings`.

Then, `updateNewTrackSettings` should move to simply `updateTrackSettings`.

I'd actually prefer to do away from putting settings into these arbitrary groups at all and have a `AnnotatorSettings` interface and a `updateSettings()` function and never have to add new symbols to the useSettings api, but this could be out of scope for this PR.

**Comment by BryonLewis** [comment]
> I'd actually prefer to do away from putting settings into these arbitrary groups at all and have a `AnnotatorSettings` interface and a `updateSettings()` function and never have to add new symbols to the useSettings api, but this could be out of scope for this PR.

Understood and I wasn't a fan either considering my last bulletpoint. I'll swap it, I was just avoiding doing an update to users who currently had data in localStorage stored a top level newTrackSettings

**Comment by subdavis** [comment]
>  I was just avoiding doing an update to users who currently had data in localStorage stored a top level newTrackSettings

On the balance, maybe don't bother?  Just change the name of the variable and let users's settings be overwritten with the defaults.   It's just a couple toggle switches.  I don't think I'd bother attempting to migrate them.

**Comment by BryonLewis** [comment]
I need to do some double checking of stuff, and I think refactoring some of the components to composition and using `toRef` so I can maintain reactivity and not have these long property chains.

**Comment by BryonLewis** [comment]
Okay I think I have this working like it should.  a `reactive` AnnotationSettings object is passed around and depending on where it is used the sub-settings are converted to a ref using `toRef` so they maintain their reactivity and can be used in vue templates while destructuring the property.
This keeps everything using the type of `AnnotationSettings` and the toRef will maintain the proper type with the wrapper of a `Ref<SubSettingsType` so it can be properly used.

**Review by subdavis — CHANGES_REQUESTED** [review]
LGTM, one nit.

File: client/dive-common/use/useModeManager.ts [hunk 1]
```diff
@@ -370,13 +378,29 @@ export default function useModeManager({
     mergeList.value = mergeList.value.filter((trackId) => !trackIds.includes(trackId));
   }
 
-  function handleRemoveTrack(trackIds: TrackId[]) {
+  async function handleRemoveTrack(trackIds: TrackId[], promptOverride = false) {
```
[the remark this answers is not in the document]
Comment (BryonLewis) [reply client/dive-common/use/useModeManager.ts:379]: are you good with `forcePromptDisable`?   just so know what you're force disabling.

[the remark this answers is not in the document]
Comment (subdavis) [reply client/dive-common/use/useModeManager.ts:379]: Yep sounds good.

**Review by subdavis — APPROVED** [review]

**Pull request merged**